indico/modules/events/notifications.py

```diff
@@ -27,14 +42,46 @@ def notify_event_creation(event, occurrences=None):
              filter(Category.notify_managers | (Category.event_creation_notification_emails != []))
              .options(joinedload('acl_entries')))
     for cat in query:
-        emails.update(cat.event_creation_notification_emails)
-        if cat.notify_managers:
-            for manager in cat.get_manager_list():
-                if manager.principal_type in (PrincipalType.user, PrincipalType.email):
-                    emails.add(manager.email)
-                elif manager.principal_type in (PrincipalType.local_group, PrincipalType.multipass_group):
-                    emails.update(x.email for x in manager.get_members())
+        emails.update(_get_emails_from_category(cat))
 
     if emails:
         template = get_template_module('events/emails/event_creation.txt', event=event, occurrences=occurrences)
         send_email(make_email(bcc_list=emails, template=template))
+
+
+def notify_move_request_creation(events, target_category, comment=''):
+    """Send email notifications when a move request is created.
+
+    :param events: List of events requested to move.
+    :param target_category: The category to move the events into.
+    :param comment: Optional requestor's comment.
+    """
+    emails = set()
+    query = (target_category.chain_query.
+             filter(Category.notify_managers | (Category.event_creation_notification_emails != []))
+             .options(joinedload('acl_entries')))
+
+    # Walk up the category chain
+    for cat in reversed([cat for cat in query]):
```

Comment:
Why reversed - won't that start with the root category now?

Reply:
The `chain_query` docstring says it returns the root first and then the intermediate categories,
so I think we need to reverse that in order to walk up from the current category

Reply:
ah good point, I was just looking at the query above but of course there the order doesn't matter so you are right...

How about this though? :)

`for cat in query.all()[::-1]`

Reply:
How about `reversed(query.all())`? I think I like that more than `[::-1]` :)

Reply:
sure